### PR TITLE
Add Blenderbot

### PR DESCRIPTION
**Pull request by jonatanklosko**
Pretty much MBart without one layernorm. I omitted `:for_causal_language_modeling` model because there is currently no checkpoint for it, so it's probably not that useful.

Waiting on `tokenizer.json` PRs to be merged [upstream](https://huggingface.co/facebook/blenderbot-400M-distill/discussions/3).

### CODE REVIEW

**Review by josevalim — APPROVED** [review]
LGTMBIHNIWIAD (Looks good to me but I have no idea what I am doing) :D

File: lib/bumblebee/text/conversation.ex [hunk 1]
```diff
@@ -47,8 +52,15 @@ defmodule Bumblebee.Text.Conversation do
...
-          sequences[[0..-1//1, input_length..-1//1]]
+
+          start_idx =
+            if encoder_decoder? do
+              1
+            else
+              Nx.axis_size(inputs["input_ids"], 1)
+            end
+
+          sequences[[0..-1//1, start_idx..-1//1]]
```
Comment (msluszniak) [inline lib/bumblebee/text/conversation.ex:63]:
```suggestion
          sequences[[.., start_idx..-1//1]]
```

File: test/bumblebee/text/blenderbot_test.exs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        "input_ids" => Nx.tensor([[1710, 86, 1085, 2]]),
+        "decoder_input_ids" => Nx.tensor([[1, 86]])
+      }
+
+      outputs = Axon.predict(model, params, inputs)
+
+      assert Nx.shape(outputs.hidden_state) == {1, 2, 1280}
+
+      assert_all_close(
+        outputs.hidden_state[[0, 0..-1//1, 1..3]],
```
Comment (msluszniak) [inline test/bumblebee/text/blenderbot_test.exs:27]:
```suggestion
        outputs.hidden_state[[0, .., 1..3]],
```

File: test/bumblebee/text/blenderbot_test.exs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        "input_ids" => Nx.tensor([[1710, 86, 1085, 2]]),
+        "decoder_input_ids" => Nx.tensor([[1, 86]])
+      }
+
+      outputs = Axon.predict(model, params, inputs)
+
+      assert Nx.shape(outputs.logits) == {1, 2, 8008}
+
+      assert_all_close(
+        outputs.logits[[0, 0..-1//1, 1..3]],
```
Comment (msluszniak) [inline test/bumblebee/text/blenderbot_test.exs:49]:
```suggestion
        outputs.logits[[0, .., 1..3]],
```

**Pull request merged**